services/ui-src/src/utils/forms/dynamicItemFields.ts

```diff
@@ -0,0 +1,114 @@
+import { AnyObject, FormJson } from "types";
+
+// first attempt to combine ilos and plans functions
+
+export const generateDrawerItemFields = (
+  form: FormJson,
+  items: AnyObject[],
+  itemType: string
+) => {
+  const fields = form.fields[0];
+  return {
+    ...form,
+    fields: [
+      {
+        ...form.fields[0],
+        props: {
+          ...form.fields[0].props,
+          choices: [
+            ...updatedItemChoiceList(
+              fields.props?.choices,
+              availableItems(items),
+              itemType
+            ),
+          ],
+        },
+      },
+    ],
+  };
+};
+
+export const parentFieldName = (itemType: string) => {
```

Comment:
I'd refactor `itemType` to `entityType` in this case, since we're differentiating between Plans and ILOS.